### PR TITLE
NuGet Audit fix - internal tools

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Follow up #3711

## What

NuGet Audit fix - internal tools

## Tests

CI

## Checklist

- ~~[ ] `CHANGELOG.md` is updated.~~
- ~~[ ] Documentation is updated.~~
- ~~[ ] New features are covered by tests.~~

### CODE REVIEW

File: tools/Directory.Packages.props [hunk 1]
```diff
@@ -2,8 +2,12 @@
   <Import Project="..\Directory.Packages.props" />
   <ItemGroup>
     <PackageVersion Include="McMaster.Extensions.CommandLineUtils" Version="4.1.1" />
-    <PackageVersion Include="Microsoft.Build" Version="17.6.3" />
-    <PackageVersion Include="NuGet.ProjectModel" Version="6.11.0" />
+    <PackageVersion Include="Microsoft.Build" Version="17.7.2" />
```
Comment (Kielek) [inline tools/Directory.Packages.props:5]: For now, it is the latest version officially supporting .NET7

File: tools/Directory.Packages.props [hunk 2]
```diff
@@ -2,8 +2,12 @@
...
+    <!-- System.Text.Json - Indirect vulnerable depenendecy https://github.com/advisories/GHSA-hh2w-p6rv-4g7w from Microsoft.Build -->
```
Comment (lachmatt) [inline tools/Directory.Packages.props:6]:
```suggestion
    <!-- System.Text.Json - Indirect vulnerable dependency https://github.com/advisories/GHSA-hh2w-p6rv-4g7w from Microsoft.Build -->
```

File: tools/Directory.Packages.props [hunk 3]
```diff
@@ -2,8 +2,12 @@
...
+    <PackageVersion Include="System.Text.Json" Version="8.0.5" />
+    <PackageVersion Include="NuGet.ProjectModel" Version="6.11.1" />
+    <!-- NuGet.ProjectModel - Indirect vulnerable depenendecy https://github.com/advisories/GHSA-447r-wph3-92pm from NuGet.ProjectModel -->
```
Comment (lachmatt) [inline tools/Directory.Packages.props:9]:
```suggestion
    <!-- NuGet.ProjectModel - Indirect vulnerable dependency https://github.com/advisories/GHSA-447r-wph3-92pm from NuGet.ProjectModel -->
```

File: tools/Directory.Packages.props (hunk 1 above)
Comment (lachmatt) [reply tools/Directory.Packages.props:5]: Maybe worth adding that as a comment in the file?

**Review by lachmatt — APPROVED** [review]

File: tools/Directory.Packages.props (hunk 1 above)
Comment (Kielek) [reply tools/Directory.Packages.props:5]: I would keep as is, we will be removing this support within ~month.

**Pull request merged**